src/geode/mesh/core/solid_mesh.cpp

```diff
@@ -456,26 +507,66 @@ namespace geode
             return *facets_;
         }
 
+        void initialize_polyhedra_around_vertex(
+            const SolidMesh< dimension >& solid )
+        {
+            polyhedra_around_vertex_ =
+                solid.vertex_attribute_manager()
+                    .template find_or_create_attribute< VariableAttribute,
+                        CachedPolyhedra >(
+                        polyhedra_around_vertex_name, CachedPolyhedra{} );
+        }
+
+        const detail::PolyhedraAroundVertexImpl& update_polyhedra_around_vertex(
+            const SolidMesh< dimension >& mesh,
+            const index_t vertex_id,
+            const absl::optional< PolyhedronVertex >& first_polyhedron ) const
+        {
+            const auto& cached = polyhedra_around_vertex_->value( vertex_id );
+            const auto& polyhedra = cached.value().polyhedra;
+            if( !cached.computed()

```

Comment:
Main question is here: when to (re)compute the value?

Reply:
I'd say the part checking if it was already computed belongs in "polyhedra_around_vertex", and update_polyhedra_around_vertex, as its name states, systematically updates it.